components/dashboard/src/admin/BlockedRepositories.tsx

```diff
@@ -0,0 +1,275 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { AdminGetListResult } from "@gitpod/gitpod-protocol";
+import { useEffect, useRef, useState } from "react";
+import { getGitpodService } from "../service/service";
+import { PageWithAdminSubMenu } from "./PageWithAdminSubMenu";
+import { BlockedRepository } from "@gitpod/gitpod-protocol/lib/blocked-repositories-protocol";
+import ConfirmationModal from "../components/ConfirmationModal";
+import Modal from "../components/Modal";
+import CheckBox from "../components/CheckBox";
+
+export function BlockedRepositories() {
+    return (
+        <PageWithAdminSubMenu title="Blocked Repositories" subtitle="Search and manage all blocked repositories.">
+            <BlockedRepositoriesList />
+        </PageWithAdminSubMenu>
+    );
+}
+
+type NewBlockedRepository = Pick<BlockedRepository, "urlRegexp" | "blockUser">;
+
+interface Props {}
+
+export function BlockedRepositoriesList(props: Props) {
+    const [searchResult, setSearchResult] = useState<AdminGetListResult<BlockedRepository>>({ rows: [], total: 0 });
+    const [queryTerm, setQueryTerm] = useState("");
+    const [searching, setSearching] = useState(false);
+
+    const [isAddModalVisible, setAddModalVisible] = useState(false);
+    const [isDeleteModalVisible, setDeleteModalVisible] = useState(false);
+
+    const [currentBlockedRepository, setCurrentBlockedRepository] = useState({
+        id: 0,
+        urlRegexp: "",
+        blockUser: false,
+    } as BlockedRepository);
+
+    const search = async () => {
+        setSearching(true);
+        try {
+            const result = await getGitpodService().server.adminGetBlockedRepositories({
+                limit: 100,
+                orderBy: "urlRegexp",
+                offset: 0,
+                orderDir: "asc",
+                searchTerm: queryTerm,
+            });
+            setSearchResult(result);
+        } finally {
+            setSearching(false);
+        }
+    };
+    useEffect(() => {
+        search(); // Initial list
+    }, []);
+
+    const add = () => {
+        setCurrentBlockedRepository({
+            id: 0,
+            urlRegexp: "",
+            blockUser: false,
+        } as BlockedRepository);
```

Comment:
Same question about using a type when using the `useState` hook to get rid of the `as` here?

Reply:
Same as above.